### ISSUE TITLE
refresh share links: so past viewers don't have access

### ISSUE THREAD

**Issue by alexfurm7** [excalidraw/excalidraw#8103]
I wish there was a way to reset the share link; to invalidate the prior link and create a new one. Currently, I need to create a duplicate of the entire drawing if I want to restrict access to viewers.

**Comment by dwelle** [comment]
Is this in the context of Excalidraw+ or excalidraw.com?

**Comment by alexfurm7** [comment]
I'm in the plus version

**Comment by dwelle** [comment]
Yes, it's something we are considering but there are some things to figure out first. One is technical (will be solved soon), the second one is a design concern.

Currently the link is the same as the drawing's, which is intentional as it allows you to simply share the url of the drawing — this is standard when sharing documents and drawings in other apps, so people expect this.

From experience, custom links are confusing and you often share the url link by mistake even if you're aware of this (happens to me all the time).

**Comment by dwelle** [comment]
> Maybe this can be shifted as read-only, presentation slides rather than full access to the actual drawing

Yes, you can create embeddable links (they're called embeddable, but the really are just readonly links) where each link has its own url and can be removed later.

**Comment by alexfurm7** [comment]
Embed links solve my needs. Thank you. Might just be a matter of wordsmithing to make it clearer how they can be used. To me, what you call embedded links provides fine grained controls over content / access, and give me a way to offer read-only privilege. Would be great if you could add an option so the links auto expire, too.

**Comment by quadcom** [comment]
From my perspective, this would require some form of user authentication. You would want to keep the URL constant for the sake of consistency. However, to restrict one person's access while retaining that of others would require identifying the person uniquely. 

I was actually hoping for the ability to authenticate users. While I am (gratefully) self-hosting Excalidraw, it means that anyone scanning domains will be able to access the application and use it as they wish. Additionally, since there is no authentication, hacking attempts are not prevented in any way.

+1 for user accounts and authentication.
